manageability-and-operations/observability-and-manageability/oci-monitoring/custom-metrics/custom-metric-python-SDK-services-limit-monitoring/files/Scripts/postServiceLimitsMetricsIAM.py

```diff
@@ -0,0 +1,300 @@
+
+# We gather the list of availability domains in the region
+identity_client = oci.identity.IdentityClient(config)
+list_availability_domains_response = identity_client.list_availability_domains(compartment_id = compartment_ocid)
+
+# Get the data from response
+print(list_availability_domains_response.data)
+
+# Initialize service client with default config file
+monitoring_client = oci.monitoring.MonitoringClient(config,service_endpoint="https://telemetry-ingestion.eu-frankfurt-1.oraclecloud.com")
+
+# Get Service Limits
+
+# Initialize service client with default config file
+limits_client = oci.limits.LimitsClient(config)
+
+# Send the request to service, some parameters are not required, see API
+# doc for more info
+list_limit_definitions_response = limits_client.list_limit_definitions(
+    compartment_id = compartment_ocid,
+    sort_by="name",
+    sort_order="ASC")
+
+# We iterate the list of all the service limits
+for x in list_limit_definitions_response.data:
+    limit = json.loads(str(x))
+    s_name = limit["service_name"]
+    l_name = limit["name"]
+    l_scope = limit["scope_type"]
+
+    print ("Service: ", s_name, "Limit: ", l_name, "Scope: ", l_scope)
+    
+    # If the resource limit has an AD scope, we have to specify the AD or we'll get an API 400 response
+    if l_scope == "AD" :
+        # We have AD scope, we've to gather the limit for all the ADs in the region and include the availabilityDomain
+        for AD in list_availability_domains_response.data:
+            
+            a_domain = json.loads(str(AD))
+            print("Availability Domain: ", a_domain)
+
+            # We gather the service limit usage
+            get_resource_availability_response = limits_client.get_resource_availability(
+            service_name = s_name,
+            limit_name = l_name,
+            compartment_id = compartment_ocid,
+            availability_domain = a_domain["name"])
+            usage = json.loads(str(get_resource_availability_response.data))
+            used = usage["used"]
+            available = usage["available"]
+
+            # Get the data from response
+            print(get_resource_availability_response.data)
+
+            # We need to gather the service limit limit
+            list_limit_values_response = limits_client.list_limit_values(
+            compartment_id = compartment_ocid,
+            service_name = s_name,
+            availability_domain = a_domain["name"],
+            limit = 1)
+            limit_limit = json.loads(str(list_limit_values_response.data[0]))
+            max_limit = limit_limit["value"]
+
+            # Get the timestamp for setup the monitoring metric post information    
+            times_stamp = datetime.datetime.now(timezone('UTC'))
+
+            # Posting custom metrics to oci monitoring for each of the metrics (max, used, available)
+
+            # Max limit
+            post_metric_data_response = monitoring_client.post_metric_data(
```

Comment:
This code is a duplicate (essentially six times) with a single change (name). Create a function for that and provide the change only. Much easier to read and maintain.